packages/js/src/core/sign-in.ts

```diff
@@ -0,0 +1,59 @@
+import qs from 'query-string';
+
+export type SignInUriParameters = {
+  authorizationEndpoint: string;
+  clientId: string;
+  redirectUri: string;
+  codeChallenge: string;
+  state: string;
+  scope?: string | string[];
+  resource?: string | string[];
+};
+
+enum ResponseType {
+  code = 'code',
+}
+
+const joinStrings = (stringOrStringArray?: string | string[]) =>
+  stringOrStringArray === undefined
+    ? ''
+    : Array.isArray(stringOrStringArray)
+    ? stringOrStringArray.join(' ')
+    : stringOrStringArray;
+
+const wrapInStringArray = (stringOrStringArray?: string | string[]): string[] =>
+  stringOrStringArray === undefined
+    ? []
+    : Array.isArray(stringOrStringArray)
+    ? stringOrStringArray
+    : [stringOrStringArray];
+
+export const generateSignInUri = ({
```

Comment:
per [discussion oflline](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1642498392221400) and convention we should ensure `openid` & `offline_access` is contained by scope.

Reply:
Fixed, PTAL.